src/resolvers/get-fuel-type.js

```diff
@@ -2,24 +2,66 @@ const Make = require('../constants/make')
 const Model = require('../constants/model')
 const FuelType = require('../constants/fuel-type')
 const getModel = require('./get-model')
+const getMake = require('./get-make')
 
-const getFuelTypeFromDescription = description => {
-  if (description.match(/e-tron/i) && description.match(/tdi/i)) {
-    return FuelType.HYBRID_DIESEL
+/**
+ * This method retrieves the fuel type for a vehicle information such
+ * as description and/or vin
+ *
+ * @param {Object} vehicle
+ * @param {string} vehicle.vin
+ * @param {string} vehicle.name
+ * @returns {Fuels|null}
+ */
+const getFuelTypeFromVehicleInfo = vehicle => {
+  const description = vehicle.name
+  if (description === 'SKODA OCTAVIA A8 COMBI Style 1ED 1,0 eTSI 110') {
+    console.log('')
+  }
+  const make = getMake(vehicle)
+  switch (make) {
+    case Make.AUDI:
+      if (description.match(/e-tron/i) && description.match(/tdi/i)) {
+        return FuelType.HYBRID_DIESEL
+      }
+      if (description.match(/((?<!(a3))a3.+e-tron|(?<!(q7))q7.+e-tron)/i)) {
+        return FuelType.HYBRID
+      }
+      if (description.match(/e-tron/i)) {
+        return FuelType.ELECTRIC
+      }
+      break
+    case Make.VOLKSWAGEN:
+      if (description.match(/gte/i)) {
+        return FuelType.HYBRID
+      }
+      if (description.match(/(e-golf|e-up!|e-up( |$)|id\.[3-5]|e-crafter)/i)) {
+        return FuelType.ELECTRIC
+      }
+      break
+    case Make.SKODA:
+      if (description.match(/citigoe/i)) {
+        return FuelType.ELECTRIC
+      }
+      if (description.match(/ iV /i)) {
```

Comment:
Might as well keep it lowercase as you ignore casing:

```suggestion
      if (description.match(/ iv /i)) {
```